kasa/interfaces/lightpreset.py

```diff
@@ -0,0 +1,89 @@
+"""Module for LightPreset base class."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Sequence
+
+from ..feature import Feature
+from ..module import Module
+from .light import LightState
+
+
+class LightPresetInfo(ABC):
+    """Light configuration preset."""
+
+    brightness: int
+
+    hue: int | None
+    saturation: int | None
+    color_temp: int | None
+
+
+class LightPreset(Module):
+    """Base interface for light preset module."""
+
+    PRESET_NOT_SET = "Not set"
+
+    def _initialize_features(self):
+        """Initialize features."""
+        device = self._device
+        self._add_feature(
+            Feature(
+                device,
+                id="light_preset",
+                name="Light preset",
+                container=self,
+                attribute_getter="preset",
+                attribute_setter="set_preset",
+                category=Feature.Category.Config,
+                type=Feature.Type.Choice,
+                choices_getter="preset_list",
+            )
+        )
+
+    @property
+    @abstractmethod
+    def preset_list(self) -> list[str]:
+        """Return built-in effects list.
+
+        Example:
+            ['Off', 'Preset 1', 'Preset 2', ...]
+        """
+
+    @property
+    @abstractmethod
+    def preset_states_list(self) -> Sequence[LightState]:
+        """Return built-in effects list.
+
+        Example:
+            ['Off', 'Preset 1', 'Preset 2', ...]
+        """
+
+    @property
+    @abstractmethod
+    def preset(self) -> str:
+        """Return current preset name."""
+
+    @abstractmethod
+    async def set_preset(
+        self,
+        preset_name: str,
+    ) -> None:
+        """Set a light preset for the device.
+
+        The device doesn't store an active effect while not enabled so store locally.
+        """
+
+    @abstractmethod
+    async def save_preset(
+        self,
+        preset_name: str,
+        preset_info: LightState,
+    ) -> None:
+        """Update the preset with preset_name with the new preset_info."""
```

Comment:
```suggestion
        """Update the preset with *preset_name* with the new *preset_info*."""
```
I'm not (either) used to it, but I think the proper way to refer parameters in a docstring is to surround it with *, for example: https://github.com/python/cpython/blob/3.12/Lib/socket.py#L810

Reply:
Done